### PR TITLE
Mongo db team

### PR DESCRIPTION
**Pull request by Shira-Abrams**
ready server side !!

### CODE REVIEW

File: server side/app.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import express from 'express';
+import morgan from 'morgan'
+import cors from 'cors'
+import PreferenceRouter from './router/preference.router.js';
+import WebsitesRouter from './router/websites.router.js';
+import profileRouter from './router/profile.router.js'
+import visitedWebsiteRouter from './router/visitedWebsite.router.js'
+import userRouter from './router/user.router.js'
+import {pageNotFound,serverErrors} from './middleware/handleErrors.js'
+import dotenv from 'dotenv'
```
Comment (SariGlick) [inline server side/app.js:11]: move to the section of external modules imports

File: server side/app.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+dotenv.config();
+connecMongo();
+app.get('/',(req,res)=>{
+    res.send('welcome to time out ');
+})
+app.use('/uploads',express.static('uploads'))
+app.use('/preference',PreferenceRouter);
+app.use('/websites',WebsitesRouter);
+app.use('/profile',profileRouter);
+app.use('/vistedwebsite',visitedWebsiteRouter);
```
Comment (SariGlick) [inline server side/app.js:30]: change the rout to camelCase, /visitedWebsite

File: server side/app.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+app.use('/user',userRouter);
+app.use(pageNotFound);
+app.use(serverErrors)
+let port= process.env.PORT;
+
+app.listen(port,()=>{
+    console.log(` runnig at http://localhost:${port}`);
```
Comment (SariGlick) [inline server side/app.js:37]: fix to "running"

File: server side/controllers/preference.controller.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+export const getAllPreferenecs=async(req,res,next)=>{
+     try {
+        const allPreferenecs=await  Preferenec.find().select('-__v');
+        return res.send(allPreferenecs);
+     } catch (error) {
+      return next({message:error.message})
+     }
+};
+export const getPreferenecById=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/preference.controller.js:12]: fix the word to preference

File: server side/controllers/preference.controller.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+             if(!updatedPreferenec)
+                return next({message:'Preferencs not found !!',status:404});
+            const newPreferenc= await Preferenec.findByIdAndUpdate(id,req.body,{new:true});
+            return res.json(newPreferenc);
+        } catch (error) {
+            next({message:error})
+        }
+};
+
+export const addPreferenec=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/preference.controller.js:47]: fix the word to preference

File: server side/controllers/preference.controller.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+    else{
+        next({message:'id is not valid'});
+    }
+    
+
+}
+
+export const updatePreferenec=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/preference.controller.js:30]: fix the word to preference

File: server side/controllers/preference.controller.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+     const newPreferenc= new Preferenec(req.body);
+     await  newPreferenc.save();
+     return res.json(newPreferenc).status(201);
+  } catch (error) {
+    return next({message:error.message})
+  }
+};
+
+
+export const deletePreferenec=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/preference.controller.js:59]: I see it needs to be fixed everywhere... so search for all the occurances of "preferenec" and replace it to "preference"

File: server side/controllers/user.controller.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  try {
+    const users = await userModel.find().populate('visitsWebsites profiles preferences');
+    res.status(200).send(users);
+  } catch (err) {
+    console.error(err);
+    res.status(500).send('Error retrieving users');
+  }
+};
+
+export const getUserId = async (req, res) => {
```
Comment (SariGlick) [inline server side/controllers/user.controller.js:16]: fix to getUserById

File: server side/controllers/websites.controller.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+export const getAllWebsites=async(req,res,next)=>{
+    try {
+     const allWebSites= await Websites.find().select('-__v');
+     return res.json(allWebSites);
+    } catch (error) {
+        next({message:error.message});
+    }
+   
+}
+export const getWebsitesById=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/websites.controller.js:13]: remove s from websites

File: server side/controllers/websites.controller.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    try {
+        const websitesBYId= await Websites.findById(id,{__v:false});
+    return res.json(websitesBYId);
+
+    } catch (error) {
+       return  next({message:error.message});
+    }
+    
+}
+export const UpdateWebSites=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/websites.controller.js:26]: remove s from websites

File: server side/controllers/websites.controller.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        const prevWebSites=await Websites.findById(id);
+        if(!prevWebSites)
+            return next({message:'websites is not found !!'})
+         const  newWebsite=await Websites.findByIdAndUpdate(id, req.body,{new:true})
+         return res.json(newWebsite).status(201);
+    } catch (error) {
+        return next({message:error.message});
+    }
+}
+export const addWebSites=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/websites.controller.js:40]: remove s from websites

File: server side/controllers/websites.controller.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        const newWebsite= new Websites(req.body);
+        await newWebsite.save();
+        return res.json(newWebsite).status(201);
+      
+    } catch (error) {
+       return  next({message:error.message});
+    }
+   
+};
+export const deleteWebsites=async(req,res,next)=>{
```
Comment (SariGlick) [inline server side/controllers/websites.controller.js:51]: remove s from websites

File: server side/router/preference.router.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import  express from 'express';
+import  {Upload}from '../middleware/uploadFiles.js';
+import  {getAllPreferenecs,getPreferenecById,updatePreferenec,deletePreferenec,addPreferenec} from '../controllers/preference.controller.js'
+
+const router=express.Router();
+router.get('/getAllPreferenecs',getAllPreferenecs);
+router.get('/getPreferenecById/:id',getPreferenecById);
+router.post('/addPreferenec',Upload.single('soundVoice'),addPreferenec);
+router.put('/updatePreferenec/:id',Upload.single('soundVoice'),updatePreferenec);
+router.delete('/deletePreferenec/:id',deletePreferenec);
```
Comment (SariGlick) [inline server side/router/preference.router.js:10]: remove the create-update-delete words, look in the user router for the correct way to do it.

File: server side/router/profile.router.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import express from 'express'
+import {getAllProfiles,getProfileById,createProfile,deleteProfile,updateProfile} from '../controllers/profileController'
+const router=express.Router();
+router.get('/getAllProfiles',getAllProfiles);
+router.get('/getProfileById/:id',getProfileById);
+router.post('/createProfile',createProfile);
+router.delete('/deleteProfile/:id',deleteProfile);
+router.put('/updateProfile/:id',updateProfile);
```
Comment (SariGlick) [inline server side/router/profile.router.js:8]: remove the create-update-delete words, look in the user router for the correct way to do it.

File: server side/router/visitedWebsite.router.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import express from 'express'
+import {getAllVisitedWebsites,getVisitedWebsiteById,deleteVisitedWebsite,updateVisitedWebsite,createVisitedWebsite} from '../controllers/visitedWebSiteController'
+const router=express.Router();
+router.get('/getAllVisitedWebsites',getAllVisitedWebsites)
+router.get('/getVisitedWebsiteById/:id',getVisitedWebsiteById)
+router.post('/createVisitedWebsite',createVisitedWebsite)
+router.put('/updateVisitedWebsite/:id',updateVisitedWebsite)
+router.delete('/deleteVisitedWebsite/:id',deleteVisitedWebsite)
```
Comment (SariGlick) [inline server side/router/visitedWebsite.router.js:8]: remove the create-update-delete words, look in the user router for the correct way to do it.

File: server side/router/websites.router.js [hunk 16]
```diff
@@ -0,0 +1,10 @@
+import  express from 'express';
+import { getWebsitesById,getAllWebsites,addWebSites,UpdateWebSites,deleteWebsites} from '../controllers/websites.controller.js';
+ const router=express.Router();
+router.get('/getAllWebsites',getAllWebsites);
+router.get('/getWebsitesById/:id',getWebsitesById);
+router.put('/UpdateWebSites/:id',UpdateWebSites);
+router.post('/addWebSites',addWebSites);
+router.delete('/deleteWebsites/:id',deleteWebsites);
```
Comment (SariGlick) [inline server side/router/websites.router.js:8]: remove the create-update-delete words, look in the user router for the correct way to do it.

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Review by SariGlick — APPROVED** [review]

**Review by SariGlick — APPROVED** [review]

**Review by moriya2004 — APPROVED** [review]

**Review by BatshevaTavger — APPROVED** [review]

**Review by oritb488 — APPROVED** [review]

**Pull request merged**